### PR TITLE
Allow image objects as background images

### LINKED ISSUE #1379: Allow Images for color_using_background_image
## Enhancement proposal
`VMobject.color_using_background_image()` can only work with image file paths. This method should also be able to handle actual `Image` objetcs. This would make working with generated Images a lot easier.

### PR DESCRIPTION
**Pull request by AntonBallmaier**
## Motivation
see #1379
closes #1379

## Explanation for Changes
The main change I made was to add a differentiation to `Camera.get_background_array()` to be able to work with image file paths as well as `Image` objects. The other changes are for the most part renamings and type hints

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [x] My new functions/classes either have a docstring or are private
- [x] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

**Comment by Nikhil0504** [comment]
Satna

**Comment by Nikhil0504** [comment]
Link #1379 to close it

**Comment by AntonBallmaier** [comment]
I don't know how to do that 😅

**Comment by kolibril13** [comment]
> I don't know how to do that 😅

image

File: manim/camera/camera.py [hunk 1]
```diff
@@ -1208,30 +1207,33 @@ def resize_background_array_to_match(self, background_array, pixel_array):
         mode = "RGBA" if pixel_array.shape[2] == 4 else "RGB"
         return self.resize_background_array(background_array, width, height, mode)
 
-    def get_background_array(self, file_name):
+    def get_background_array(self, image: Union[Image.Image, str]):
```
Comment (kolibril13) [inline manim/camera/camera.py:1210]: what about `typing.Union[Image.Image,str, pathlib.Path]` ?
But I did not test if pathlib works here.

**Comment by Nikhil0504** [comment]
> > I don't know how to do that 😅
> 
> image

I think he means he doesn't knows what to do to solve this.

File: manim/camera/camera.py (hunk 1 above)
Comment (AntonBallmaier) [reply manim/camera/camera.py:1210]: They should works since they are immutable and hashable.

**Comment by AntonBallmaier** [comment]
> > I don't know how to do that sweat_smile
> 
> image

I don't seem to have the rights to to that...

**Comment by jsonvillanueva** [comment]
In the future, you can also add `Fixes #XYZ` in your inital PR body and it'll auto link the issue referred to..

File: manim/camera/camera.py (hunk 1 above)
Comment (kolibril13) [reply manim/camera/camera.py]: Perfect.
I will try to review and test it later today, but maybe also only next week.

File: manim/camera/camera.py [hunk 2]
```diff
@@ -1208,30 +1208,33 @@ def resize_background_array_to_match(self, background_array, pixel_array):
...
-    def get_background_array(self, file_name):
+    def get_background_array(self, image: Union[Image.Image, pathlib.Path, str]):
         """Gets the background array that has the passed file_name.
 
         Parameters
         ----------
-        file_name : str
-            The file_name of the background image.
+        image
+            The background image or its file_name.
```
Comment (behackl) [inline manim/camera/camera.py:1217]:
```suggestion
            The background image or its file name.
```

File: manim/camera/camera.py [hunk 3]
```diff
@@ -1208,30 +1208,33 @@ def resize_background_array_to_match(self, background_array, pixel_array):
...
 
         Returns
         -------
         np.ndarray
-            The pixel array of the file whose file name is `file_name`
+            The pixel array of the given image (file)
```
Comment (behackl) [inline manim/camera/camera.py:1222]:
```suggestion
            The pixel array of the image.
```

**Review by behackl — APPROVED** [review]
Two minor suggestions regarding documentation, otherwise this is good to go.

**Pull request merged**